notebooks/405-paddleOCR-webcam/405-paddleOCR-webcam.ipynb

```diff
@@ -0,0 +1,557 @@
+    "    4. Visualize the results.\n",
+    "\n",
+    "    Parameters:\n",
+    "        source: the webcam number to feed the video stream with primary webcam set to \"0\", or the video path.  \n",
+    "        flip: to be used by VideoPlayer function for flipping capture image\n",
+    "        use_popup: False for showing encoded frames over this notebook, True for creating a popup window.\n",
+    "        skip_first_frames: Number of frames to skip at the beginning of the video. \n",
+    "    \"\"\"\n",
+    "    # create video player to play with target fps\n",
+    "    player = None\n",
+    "    try:\n",
+    "        player = utils.VideoPlayer(source=source, flip=flip, fps=30, skip_first_frames=skip_first_frames)\n",
+    "        # Start video capturing\n",
+    "        player.start()\n",
+    "        if use_popup:\n",
+    "            title = \"Press ESC to Exit\"\n",
+    "            cv2.namedWindow(winname=title, flags=cv2.WINDOW_GUI_NORMAL | cv2.WINDOW_AUTOSIZE)\n",
+    "\n",
+    "        processing_times = collections.deque()\n",
+    "        while True:\n",
+    "            # grab the frame\n",
+    "            frame = player.next()\n",
+    "            if frame is None:\n",
+    "                print(\"Source ended\")\n",
+    "                break\n",
+    "            # if frame larger than full HD, reduce size to improve the performance\n",
+    "            scale = 1280 / max(frame.shape)\n",
+    "            if scale < 1:\n",
+    "                frame = cv2.resize(src=frame, dsize=None, fx=scale, fy=scale,\n",
+    "                                   interpolation=cv2.INTER_AREA)\n",
+    "            # preprocess image for text detection\n",
+    "            test_image = image_preprocess(frame,640)\n",
+    "                \n",
+    "            # measure processing time for text detection\n",
+    "            start_time = time.time()\n",
+    "            # perform the inference step\n",
+    "            det_results = det_compiled_model([test_image])[det_output_layer]\n",
+    "            stop_time = time.time()\n",
+    "\n",
+    "            # Postprocessing for Paddle Detection\n",
+    "            ori_im = frame.copy()\n",
+    "            data = {'image': frame}\n",
+    "            data_resize = processing.DetResizeForTest(data)\n",
+    "            data_list = []\n",
+    "            keep_keys = ['image', 'shape']\n",
+    "            for key in keep_keys:\n",
+    "                data_list.append(data_resize[key])\n",
+    "            img, shape_list = data_list\n",
+    "\n",
+    "            shape_list = np.expand_dims(shape_list, axis=0) \n",
+    "            pred = det_results[0]    \n",
+    "            if isinstance(pred, paddle.Tensor):\n",
+    "                pred = pred.numpy()\n",
+    "            segmentation = pred > 0.3\n",
+    "\n",
+    "            boxes_batch = []\n",
+    "            for batch_index in range(pred.shape[0]):\n",
```

Comment:
This is a very long function now. It could be beneficial if you can separate some parts.